libs/saml2-web-sso/src/Text/XML/DSig.hs

```diff
@@ -236,32 +250,150 @@ verifyRoot creds el = do
 
 -- | Try a list of creds against a document.  If all fail, return a list of errors for each cert; if
 -- *any* succeed, return the empty list.
-verifyIO :: NonEmpty SignCreds -> LBS -> String -> IO [(SignCreds, Either HS.SignatureError ())]
-verifyIO creds el signedID = capture' $ do
-  results <- NL.zip creds <$> forM creds (\cred -> verifyIO' cred el signedID)
+verifyIO :: NonEmpty SignCreds -> LBS -> String -> IO (SignCreds, Either HS.SignatureError HXTC.XmlTree)
+verifyIO creds el sid = capture' $ do
+  results <- NL.zip creds <$> forM creds (\cred -> verifyIO' cred el sid)
   case NL.filter (isRight . snd) results of
-    (_ : _) -> pure []
-    [] -> pure $ NL.toList results
+    [result] -> pure result
+    _ -> throwIO . ErrorCall $ "all credentials failed to verify signature"
   where
     capture' :: IO a -> IO a
     capture' action =
       hCapture [stdout, stderr] action >>= \case
         ("", out) -> pure out
         (noise, _) -> throwIO . ErrorCall $ "noise on stdout/stderr from hsaml2 package: " <> noise
 
-verifyIO' :: SignCreds -> LBS -> String -> IO (Either HS.SignatureError ())
-verifyIO' (SignCreds SignDigestSha256 (SignKeyRSA key)) el signedID = runExceptT $ do
+verifyIO' :: SignCreds -> LBS -> String -> IO (Either HS.SignatureError HXTC.XmlTree)
+verifyIO' (SignCreds SignDigestSha256 (SignKeyRSA key)) el sid = runExceptT $ do
```

Comment:
IIRC we agreed that the use of prime is an antipattern. Can we come up with a more meaningful name?

Reply:
that's a big ask if you want to fix it consistently :-) i just kept the name here, but fixed it in some places.  is it ok if we stop here for now?

Reply:
(note that this function was here before, it wasn't introduced or renamed in this PR.)